x-pack/plugins/synthetics/public/apps/synthetics/hooks/use_monitor_alert_enable.tsx

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { i18n } from '@kbn/i18n';
+import { ConfigKey, EncryptedSyntheticsMonitor } from '../components/monitors_page/overview/types';
+import { enableMonitorAlertAction, selectMonitorUpsertStatuses } from '../state';
+
+export interface EnableStateMonitorLabels {
+  failureLabel: string;
+  enabledSuccessLabel: string;
+  disabledSuccessLabel: string;
+}
+
+export function useMonitorAlertEnable({ reloadPage }: { reloadPage?: () => void }) {
+  const dispatch = useDispatch();
+  const upsertStatuses = useSelector(selectMonitorUpsertStatuses);
+  const alertStatus = useCallback(
+    (configId: string) => upsertStatuses[configId]?.alertStatus,
+    [upsertStatuses]
+  );
+
+  const updateAlertEnabledState = useCallback(
+    ({
+      monitor,
+      name,
+      configId,
+    }: {
+      monitor: Partial<EncryptedSyntheticsMonitor>;
+      configId: string;
+      name: string;
+    }) => {
+      dispatch(
+        enableMonitorAlertAction.get({
+          id: configId,
+          monitor,
+          success: {
+            message: monitor[ConfigKey.STATUS_ALERT_ENABLED]
+              ? enabledSuccessLabel(name)
+              : disabledSuccessLabel(name),
+            lifetimeMs: 3000,
+            testAttribute: 'uptimeMonitorAlertUpdateSuccess',
+          },
+          error: {
+            message: {
+              title: enabledFailLabel(name),
+            },
+            lifetimeMs: 10000,
+            testAttribute: 'uptimeMonitorAlertEnabledUpdateFailure',
+          },
+        })
+      );
+      if (reloadPage) reloadPage();
+    },
+    [dispatch, reloadPage]
+  );
+
+  return { updateAlertEnabledState, alertStatus };
+}
+
+const enabledSuccessLabel = (name: string) =>
+  i18n.translate('xpack.synthetics.overview.actions.enabledSuccessLabel.alert', {
+    defaultMessage: 'Alert for Monitor "{name}" enabled successfully',
```

Comment:
```suggestion
    defaultMessage: 'Alert for monitor "{name}" enabled successfully',
```
Don't think this is supposed to be capitalized @florent-leborgne